datalad_next/constraints/git.py

```diff
@@ -66,3 +69,69 @@ def short_description(self):
             '(single-level) ' if self._allow_onelevel else '',
             ' or refspec pattern' if self._refspec_pattern else '',
         )
+
+
+class EnsureRemoteName(EnsureGitRefName):
+    """Ensures a remote name is provided, and if it fulfills optional
+    requirements"""
+
+    def __init__(self,
+                 existing: bool | None = None,
+                 dsarg: DatasetParameter | None = None):
+        """
+        Parameters
+        ----------
+        existing: bool
```

Comment:
This is just nitpicking ;-) 

I find the name `existing` and the logic of `True` and `False` slightly non-intuitive. Because to me I would think that `existing = True` means: "check for existing", and `existing = False` means: "don't check for existing".

But here `existing = False` means: "don't check for existing and check for not existing". And `existing = True` means: "don't check for not existing, but check for existing". I see the value in the checks. I understand that they are mutually exclusive. 

An (obviously useless) name that reflects the semantics seems to be: `if_not_none_and_true_check_existence_if_not_none_and_false_check_none_existance`.

An option might be to separate the request for checks and the type of check. E.g.,

- perform_check: bool: check the validation criteria if True, else don't check it.
- validation_criteria: str: one of "existance", "non-existance": check for .... if `perform check` is True.

As I said "nitpicking" ;-)

Reply:
(nitpick 2.0) another option would be a parameter named `validation` with values "nothing", "existence", or "non-existence".

Reply:
alternative suggestion: what about ``pre-exists``?

Reply:
ping